Add support for Confluent Schema registry and avro serdes (#51)
logstash-plugins/logstash-integration-kafka: docs/input-kafka.asciidoc

| <<plugins-{type}s-{plugin}-sasl_jaas_config>> |<<string,string>>|No
| <<plugins-{type}s-{plugin}-sasl_kerberos_service_name>> |<<string,string>>|No
| <<plugins-{type}s-{plugin}-sasl_mechanism>> |<<string,string>>|No
| <<plugins-{type}s-{plugin}-schema_registry_key>> |<<string,string>>|No

> | <<plugins-{type}s-{plugin}-schema_registry_proxy>> |<<string,uri>>|No

Reviewer: ```suggestion
| <<plugins-{type}s-{plugin}-schema_registry_proxy>> |<<uri,uri>>|No
```

| <<plugins-{type}s-{plugin}-schema_registry_secret>> |<<string,string>>|No

> | <<plugins-{type}s-{plugin}-schema_registry_url>> |<<string,uri>>|No

Reviewer: ```suggestion
| <<plugins-{type}s-{plugin}-schema_registry_url>> |<<uri,uri>>|No
```

* Value type is <<string,string>>
* There is no default value for this setting.

Set the username for basic authorization to access remote Schema Registry.

[id="plugins-{type}s-{plugin}-schema_registry_proxy"]
===== `schema_registry_proxy`

* Value type is <<uri,uri>>
* There is no default value for this setting.

Set the address of a forward HTTP proxy. An empty string is treated as if proxy was not set.

[id="plugins-{type}s-{plugin}-schema_registry_secret"]
===== `schema_registry_secret`

* Value type is <<string,string>>
* There is no default value for this setting.

Set the password for basic authorization to access remote Schema Registry.

[id="plugins-{type}s-{plugin}-schema_registry_url"]
===== `schema_registry_url`

* Value type is <<uri,uri>>

This is the URI that points to an instance of https://docs.confluent.io/current/schema-registry/index.html[Schema Registry] service,
used to manage Avro schemas. When the Schema Registry is configured it's avoided to specify also the <<plugins-{type}s-{plugin}-value_deserializer_class>>.
On the Schema Registry service must be previously uploaded the Avro schemas that are used to deserialize the data from the specified topics.

> Those schemas must follow a naming convention with the pattern `<topic name>-value`.

Reviewer: Please consider this rewording: 

The URI that points to an instance of the
https://docs.confluent.io/current/schema-registry/index.html[Schema Registry]
service, used to manage Avro schemas. 
Be sure that the Avro schemas for deserializing the data from the specified
topics have been uploaded to the Schema Registry service. 
The schemas must follow a naming convention with the pattern `<topic
name>-value`.

Use either the Schema Registry config option or the
<<plugins-{type}s-{plugin}-value_deserializer_class>> config option, but not
both.

Reviewer: Please make sure that there is a return between the last line of this description and the next options anchor.

  * Value type is <<string,string>>
  * Default value is `"org.apache.kafka.common.serialization.StringDeserializer"`

> Java Class used to deserialize the record's value. This settings conflicts with <<plugins-{type}s-{plugin}-schema_registry_url>>.
> A custom value deserializer could be used only if not using a Schema Registry.

Reviewer: ```suggestion
Java Class used to deserialize the record's value. 

A custom value deserializer can be used only if you are not using a Schema Registry.
Use either the value_deserializer_class config option or the
<<plugins-{type}s-{plugin}-schema_registry_url>> config option, but not both.
```

Author: Thanks @karenzone for the suggestion, integrated!
